### PR TITLE
Fix #5103: Add support for BigInt literals

### LINKED ISSUE #5103: Proposal: support for BigInt literals (`1n`)
Note: this should be tagged with `[Awaiting Stage 4]` at this time.

Node.js supports BigInt, including some niceties such as `process.hrtime.bigint()`.

This doesn't seem to mean a lot of changes for CoffeeScript, besides maybe adding BigInt literals.

### Input Code
```coffee
42n
0b101010n
0o52n
0x2an
```
### Expected Behavior
The number grammar is expanded to support the `n` suffix: https://tc39.github.io/proposal-bigint/#sec-grammar-change

### Current Behavior
`unexpected identifier` error when compiling a BigInt syntax

### Possible Solution
I don't know how CoffeeScript currently handles number literals, but essentially the textual representation of the literal should be copied over to the output: `42n` in coffee → `42n` in ecmascript.

(In the interim, using ``` `42n` ``` works as expected.)

### Context
Using native BigInt instead of external libraries, etc. in recent Node.js

### Environment
* Node.js version: 10.8.0 (apparently all 10.x versions have some form of BigInt support)

### PR DESCRIPTION
**Pull request by rdeforest**
* `src/lexer.coffee`
  * Add `/^\d+n/` as valid numeric literal syntax
  * Allow trailing `n` on binary, octal and hex literals
* `test/numbers_bigint.coffee`
  * BigInt exists
  * `42n` equals  `BigInt 42`
  * `42n` equals `0b101010n`
  * `42n` equals `0o52n`
  * `42n` equals `0x2an`
* `Cakefile`
  * skip `test/numbers/bigint.coffee` if `1n` isn't valid and true.

Tested in node v6, v8, v9 and v10.

### CODE REVIEW

**Comment by rdeforest** [comment]
I just noticed #5103 also listed binary, octal and hex input formats as valid, so I'll add those now and update the PR.

**Comment by GeoffreyBooth** [comment]
Thanks for this, though according to https://github.com/tc39/proposal-bigint and the link you posted BigInt literals are currently at Stage 3. We can keep this PR open until the feature reaches Stage 4.

**Comment by WA9ACE** [comment]
Now that the referenced PR has been merged is it considered Stage 4?

**Comment by GeoffreyBooth** [comment]
Yes, it’s reached Stage 4, and Node supports the syntax now as well. Any updates needed before this merges in? @helixbass?

File: test/numbers_bigint.coffee [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+test "BigInt exists", ->
+  'object' is typeof BigInt
+
+test "Parser recognizes decimal BigInt literals", ->
+  eq 42n, BigInt 42
+
+test "Parser recognizes binary BigInt literals", ->
+  eq 42n, 0b101010n
+
+test "Parser recognizes decimal BigInt literals", ->
```
Comment (helixbass) [inline test/numbers_bigint.coffee]: `decimal` -> `octal`

File: test/numbers_bigint.coffee [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  eq 42n, 0o52n
+
+test "Parser recognizes decimal BigInt literals", ->
```
Comment (helixbass) [inline test/numbers_bigint.coffee]: `decimal` -> `hexadecimal`

**Comment by GeoffreyBooth** [comment]
Hey, CI passed! Is this ready to merge in?

Can you please add Node 12 to `.travis.yml` and `appveyor.yml`? Just see where `10` is listed and do the same for `12` just below it. If CI still passes in 12 and no one has any other notes, I assume this is good to merge.

**Pull request merged**